go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -1321,7 +1321,10 @@ func overwriteRoutesTableID(routes []netlink.Route, tableID int) {
 }
 
 func getRouteTableID(ifIndex int) int {
-	return ifIndex + routingTableIDStart
+	if !util.IsNetworkSegmentationSupportEnabled() {
+		return ifIndex + routingTableIDStart
+	}
+	return ifIndex + int(ovnconfig.UserDefinedNetworks.VRFTableBase) + int(ovnconfig.UserDefinedNetworks.MaxNetworks)
```

Comment:
two concerns here, you might have thought about it already.

1.  how is upgrade is handled here ? Let's say we have secondary EIPs, then upgrade to new OCP version and enable user defined segmentation networks. doesn't this make existing EIP custom routing table invalid anymore ?
2.  how about node reboot scenario ? I don't know if ifIndex remains same, If it's not, it's a generic issue for vrd table id allocation based on if index, how do we fix this ?